gdnative-derive/src/methods.rs

```diff
@@ -281,17 +287,50 @@ fn impl_gdnative_expose(ast: ItemImpl) -> (ItemImpl, ClassMethodExport) {
                                                 ));
                                                 return false;
                                             }
+                                        }
+                                        // name override
+                                        "name" => {
+                                            let value = if let syn::Lit::Str(lit_str) = lit {
+                                                lit_str.value()
+                                            } else {
+                                                errors.push(syn::Error::new(
+                                                    last.span(),
+                                                    "unexpected type for name value, expected Str",
+                                                ));
+                                                return false;
+                                            };
 
+                                            // NOTE: We take advantage of rust identifiers following
+                                            // the same syntax rules as Godot method identifiers. See:
+                                            // https://docs.godotengine.org/en/stable/getting_started/scripting/gdscript/gdscript_basics.html#identifiers
```

Comment:
I wonder if it's really required that a method name is a valid identifier. We know for sure that properties can have non-identifier names: the property name of the controller of an `AnimationTree` is `"parameters/playback"`, for example. I guess a non-identifier name can still be called using `call()`? If so, is it still necessary to check this here?

Reply:
I've just checked and indeed exported methods don't need to be valid identifiers, so I've lifted the restriction and now any string can be an exported method.